components/dashboard/src/service/public-api.ts

```diff
@@ -0,0 +1,43 @@
+/**
+ * Copyright (c) 2022 Gitpod GmbH. All rights reserved.
+ * Licensed under the GNU Affero General Public License (AGPL).
+ * See License-AGPL.txt in the project root for license information.
+ */
+
+import { createConnectTransport, createPromiseClient, Interceptor } from "@bufbuild/connect-web";
+
+// Import service definition that you want to connect to.
+import { TeamsService } from "@gitpod/public-api/lib/gitpod/experimental/v1/teams_connectweb";
+import { getGitpodService } from "./service";
+
+let token: string | undefined;
+
+const authInterceptor: Interceptor = (next) => async (req) => {
+    if (!token) {
+        const newToken = await getGitpodService().server.generateNewGitpodToken({
+            type: 1,
+            scopes: [
+                "function:getGitpodTokenScopes",
+                "function:getWorkspace",
+                "function:getWorkspaces",
+                "function:createTeam",
+                "function:joinTeam",
+                "function:getTeamMembers",
+                "function:getGenericInvite",
+                "function:listenForWorkspaceInstanceUpdates",
+                "resource:default",
```

Comment:
Why is it necessary to generate a new token here? Won't the user always have a suitable token when performing this action?

Reply:
They have a cookie based session which is not compatible with the AccessToken based auth on Public API. I agree this is not great, or pretty, but it's a temporary solution until we have first class Access Tokens.